src/main/java/com/zaxxer/nuprocess/internal/BasePosixProcess.java

```diff
@@ -82,6 +92,26 @@
 
    private Pointer posix_spawn_file_actions;
 
+   // Launches threads under which changes to cwd do not affect the cwd of the process.
+   public static class LinuxCwdChangeableThreadFactory implements ThreadFactory {
+      private final AtomicInteger threadCount = new AtomicInteger(0);
+
+      @Override
+      public Thread newThread(final Runnable r) {
+         Runnable unshareCwdThenSpawn = new Runnable() {
+            @Override
+            public void run() {
+              int rc = LinuxLibC.unshare(LinuxLibC.CLONE_FS);
```

Comment:
If I recall correctly, exceptions thrown inside a Runnable are caught by
the executor, then wrapped in an ExecutionException and thrown on the other
side. We'll want to unwrap them and re-throw them.
2015年9月5日(土) 7:41 Brett Wooldridge <notifications@github.com>:

> In src/main/java/com/zaxxer/nuprocess/internal/BasePosixProcess.java
> <https://github.com/brettwooldridge/NuProcess/pull/32#discussion_r38811372>
> :
>
> > @@ -82,6 +92,26 @@
> >
> >     private Pointer posix_spawn_file_actions;
> >
> > +   // Launches threads under which changes to cwd do not affect the cwd of the process.
> > +   public static class LinuxCwdChangeableThreadFactory implements ThreadFactory {
> > +      private final AtomicInteger threadCount = new AtomicInteger(0);
> > +
> > +      @Override
> > +      public Thread newThread(final Runnable r) {
> > +         Runnable unshareCwdThenSpawn = new Runnable() {
> > +            @Override
> > +            public void run() {
> > +              int rc = LinuxLibC.unshare(LinuxLibC.CLONE_FS);
>
> Makes since on the unshare().
>
> With respect to the exception, I think it is going to be swallowed by the
> executor, so there needs to be a way to propagate it out so that it is
> eventually thrown from the start() method.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/brettwooldridge/NuProcess/pull/32/files#r38811372>.
>